src/App.tsx

```diff
@@ -1,156 +1,96 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, { useEffect, useState } from 'react';
 
-const USER_ID = 0;
+import { Header } from './components/Header';
+import { Main } from './components/Main';
+import { Footer } from './components/Footer';
+import { ErrorPanel } from './components/ErrorPanel';
+
+import { Todo } from './types/Todo';
+import { getTodos } from './api/todos';
+
+const USER_ID = 6160;
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [error, setError] = useState('');
+  const [showError, setShowError] = useState(false);
+  const [isEditing] = useState(false);
+  const [filteredTodos, setFilteredTodos] = useState<Todo[]>(todos);
+  const [selectedFilter, setSelectedFilter] = useState<
+    'all' | 'active' | 'completed'
+  >('all');
+
+  const activeTodos = todos.filter(todo => !todo.completed);
+  const completedTodos = todos.filter(todo => todo.completed);
```

Comment:
now you calculate it on each re-render. Let's improve our performance and think of a way of how we can do filtering only the way it is needed